storages/merger/list.go

```diff
@@ -0,0 +1,134 @@
+package merger
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strconv"
+
+	"github.com/allegro/akubra/log"
+	"github.com/allegro/akubra/storages/merger/s3datatypes"
+	"github.com/allegro/akubra/transport"
+)
+
+// MergeBucketListResponses unifies responses from multiple backends
+func MergeBucketListResponses(successes []transport.ResErrTuple) (resp *http.Response, err error) {
+	if len(successes) == 0 {
+		err = fmt.Errorf("No successful responses")
+		return
+	}
+	keys := objectsContainer{
+		list: make([]fmt.Stringer, 0),
+		set:  make(map[string]struct{}),
+	}
+	prefixes := objectsContainer{
+		list: make([]fmt.Stringer, 0),
+		set:  make(map[string]struct{}),
+	}
+	var listBucketResult s3datatypes.ListBucketResult
+	for _, tuple := range successes {
+		resp = tuple.Res
+		listBucketResult = extractListResults(resp)
+		keys.append(listBucketResult.Contents.ToStringer()...)
+		prefixes.append(listBucketResult.CommonPrefixes.ToStringer()...)
+	}
+
+	req := successes[0].Res.Request
+	reqQuery := req.URL.Query()
+	maxKeysQuery := reqQuery.Get("max-keys")
+	maxKeys, err := strconv.Atoi(maxKeysQuery)
+	if err != nil {
+		maxKeys = 1000
+	}
+
+	listBucketResult = pickResultSet(keys, prefixes, maxKeys, listBucketResult)
+
+	bodyBytes, err := xml.Marshal(listBucketResult)
+	if err != nil {
+		log.Debug("Problem marshalling ObjectStore response body, %s", err)
+		return nil, err
+	}
+	buf := bytes.NewBuffer(bodyBytes)
+	resp.Body = ioutil.NopCloser(buf)
```

Comment:
Same question as below, why the `resp` is not initialzied anywhere (does Go automatically initializes the struct with deafault values, when you reference any of its fields?), and what about the headers from the responses in `successes`?

Reply:
There is no need for such initialization (see answer above). 
Yes go initialises all values (not just struct types) with type zeros (0, false, "", nil). Struct is 'zero' if all fields are 'zeroes'.
Headers are just fine :)